src/Interpreters/HashJoin.cpp

```diff
@@ -68,6 +68,7 @@ namespace JoinStuff
     template <ASTTableJoin::Kind KIND, ASTTableJoin::Strictness STRICTNESS>
     void JoinUsedFlags::reinit(size_t size)
     {
+        (void)size;
```

Comment:
Maybe you are using outdated version of gcc - that's plain dangerous and better not to do any tweaks for it.

Reply:
Well, this just silence the compiler warning about unused variable.